ui/src/map/layers/alignment/preview-deleted-point-icon-features-layer.ts

```diff
@@ -0,0 +1,221 @@
+    );
+    return deletedIconFeature;
+}
+
+function createDeletedSwitchIconFeature(
+    s: LayoutSwitch,
+    structure: SwitchStructure,
+    resolution: number,
+): Feature<OlPoint> | undefined {
+    const presentationJointNumber = structure?.presentationJointNumber;
+    const presentationJoint = s.joints.find((joint) => joint.number === presentationJointNumber);
+
+    return presentationJoint
+        ? deletedSwitchFeature(presentationJoint.location, resolution <= SWITCH_LARGE_SYMBOLS)
+        : undefined;
+}
+
+const layerName: MapLayerName = 'preview-deleted-point-icon-features-layer';
+export function createDeletedPreviewPointIconFeaturesLayer(
+    mapTiles: MapTile[],
+    existingOlLayer: VectorLayer<Feature<LineString | OlPoint | Rectangle>> | undefined,
+    publicationCandidates: PublicationCandidate[],
+    designPublicationMode: DesignPublicationMode,
+    layoutContext: LayoutContext,
+    changeTimes: ChangeTimes,
+    onViewContentChanged: (items: OptionalShownItems) => void,
+    onLoadingData: (loading: boolean) => void,
+    resolution: number,
+): MapLayer {
+    const { layer, source, isLatest } = createLayer(layerName, existingOlLayer);
+
+    const layerLayoutContext =
+        designPublicationMode === 'PUBLISH_CHANGES'
+            ? officialLayoutContext(layoutContext)
+            : draftMainLayoutContext();
+
+    function updateShownSwitches(switchIds: LayoutSwitchId[]) {
+        const compare = switchIds.sort().join();
+
+        if (compare !== shownSwitchesCompare) {
+            shownSwitchesCompare = compare;
+            onViewContentChanged({ switches: switchIds });
+        }
+    }
+
+    function updateShownKmPosts(kmPostIds: LayoutKmPostId[]) {
+        const compare = kmPostIds.sort().join();
+
+        if (compare !== shownKmPostsCompare) {
+            shownKmPostsCompare = compare;
+            onViewContentChanged({ kmPosts: kmPostIds });
+        }
+    }
+
+    const deletedKmPostCandidates = publicationCandidates.filter(
+        (c) => c.type === 'KM_POST' && c.operation === 'DELETE',
+    );
+    const deletedSwitchCandidates = publicationCandidates.filter(
+        (c) => c.type === 'SWITCH' && c.operation === 'DELETE',
+    );
+
+    const deletedSwitchesPromise =
+        resolution <= SWITCH_SHOW && deletedSwitchCandidates.length > 0
+            ? Promise.all(
+                  mapTiles.map((t) =>
+                      getSwitchesByTile(changeTimes.layoutSwitch, t, layerLayoutContext),
+                  ),
+              ).then((switches) => {
+                  return switches
+                      .flat()
+                      .filter(filterUniqueById((s) => s.id))
+                      .filter((s) =>
+                          deletedSwitchCandidates.some((candidate) => candidate.id === s.id),
+                      );
+              })
+            : Promise.resolve([]);
+    const kmPostPromise =
+        resolution < SWITCH_SHOW && deletedKmPostCandidates.length > 0
```

Comment:
Olemassaoleville kilometripylväille ei ole omaa resoluutiolimittiä, sillä ne haetaan bäkkäriltä resoluution mukaan pikkuhiljaa harventaen (esim. 100km skaalalla näkyy edelleen yksittäisiä kilometripylväitä). Totesin kuitenkin että näyttää oudolta jos poistettujen kilometripylväiden ikonit näytetään liian korkealla zoomitasoilla -> sidoin niiden näyttämisen samaan zoomitasoon kuin vaihteidenkin. Korostukset sekä vaihteille että kilometripylväille näytetään kuitenkin jokaisella zoomitasolla, tämä vaikuttaa ainoastaan ikoneihin